`FloatFilter` does not work

### Steps to reproduce

Send following GraphQL query:

```gql
query getMangaUnread($mangaId: Int!, $max: Float!) {
  chapters(condition: { mangaId: $mangaId, isRead: false }, filter: { chapterNumber: { lessThanOrEqualTo: $max } }) {
    nodes {
      id
      chapterNumber
    }
  }
}
```

### Expected behavior

A result

### Actual behavior

```
{
  "errors": [
    {
      "message": "Exception while fetching data (/chapters) : argument type mismatch\r\n\r\njava.lang.IllegalArgumentException: argument type mismatch\n\tat java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:65)\n\tat java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)\n\tat java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)\n\tat kotlin.reflect.jvm.internal.calls.CallerImpl$Constructor.call(CallerImpl.kt:41)\n\tat kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:258)\n\tat kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:155)\n\tat com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.mapToKotlinObject(convertArgumentValue.kt:121)\n\tat com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.convertValue(convertArgumentValue.kt:76)\n\tat com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.convertArgumentValue(convertArgumentValue.kt:54)\n\tat com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.mapToKotlinObject(convertArgumentValue.kt:119)\n\tat com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.convertValue(convertArgumentValue.kt:76)\n\tat com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.convertArgumentValue(convertArgumentValue.kt:54)\n\tat com.expediagroup.graphql.generator.execution.FunctionDataFetcher.mapParameterToValue(FunctionDataFetcher.kt:98)\n\tat com.expediagroup.graphql.generator.execution.FunctionDataFetcher.getParameters(FunctionDataFetcher.kt:78)\n\tat com.expediagroup.graphql.generator.execution.FunctionDataFetcher.get(FunctionDataFetcher.kt:57)\n\tat suwayomi.tachidesk.graphql.directives.RequireAuthDirectiveWiring.onField$lambda$0(RequireAuthDirectiveWiring.kt:44)\n\tat graphql.execution.ExecutionStrategy.invokeDataFetcher(ExecutionStrategy.java:533)\n\tat graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:497)\n\tat graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:438)\n\tat graphql.execution.ExecutionStrategy.resolveFieldWithInfo(ExecutionStrategy.java:397)\n\tat graphql.execution.ExecutionStrategy.getAsyncFieldValueInfo(ExecutionStrategy.java:335)\n\tat graphql.execution.AsyncExecutionStrategy.execute(AsyncExecutionStrategy.java:57)\n\tat graphql.execution.Execution.executeOperation(Execution.java:181)\n\tat graphql.execution.Execution.execute(Execution.java:117)\n\tat graphql.GraphQL.execute(GraphQL.java:546)\n\tat graphql.GraphQL.lambda$parseValidateAndExecute$13(GraphQL.java:476)\n\tat java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)\n\tat java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2341)\n\tat graphql.GraphQL.parseValidateAndExecute(GraphQL.java:471)\n\tat graphql.GraphQL.lambda$executeAsync$9(GraphQL.java:429)\n\tat java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)\n\tat java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2341)\n\tat graphql.GraphQL.executeAsync(GraphQL.java:418)\n\tat com.expediagroup.graphql.server.execution.GraphQLRequestHandler.execute(GraphQLRequestHandler.kt:103)\n\tat com.expediagroup.graphql.server.execution.GraphQLRequestHandler.executeRequest$suspendImpl(GraphQLRequestHandler.kt:82)\n\tat com.expediagroup.graphql.server.execution.GraphQLRequestHandler.executeRequest(GraphQLRequestHandler.kt)\n\tat com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invokeSuspend(GraphQLServer.kt:62)\n\tat com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invoke(GraphQLServer.kt)\n\tat com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invoke(GraphQLServer.kt)\n\tat kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)\n\tat kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:43)\n\tat kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:286)\n\tat com.expediagroup.graphql.server.execution.GraphQLServer.execute$suspendImpl(GraphQLServer.kt:49)\n\tat com.expediagroup.graphql.server.execution.GraphQLServer.execute(GraphQLServer.kt)\n\tat suwayomi.tachidesk.graphql.controller.GraphQLController$execute$1$1.invokeSuspend(GraphQLController.kt:26)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)\n\tat kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)\n\tat kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)\n\tat kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)\nCaused by: java.lang.ClassCastException: Cannot cast java.lang.Double to java.lang.Float\n\tat java.base/java.lang.Class.cast(Class.java:4067)\n\tat java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)\n\t... 52 more\n",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "chapters"
      ]
    }
  ]
}
```

In readable log form:

```
22:40:22.055 [DefaultDispatcher-worker-2] ERROR suwayomi.tachidesk.graphql.server.TachideskGraphQLServer -- GraphQL execution failed due to
java.lang.IllegalArgumentException: argument type mismatch
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:65)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
        at kotlin.reflect.jvm.internal.calls.CallerImpl$Constructor.call(CallerImpl.kt:41)
        at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:258)
        at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:155)
        at com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.mapToKotlinObject(convertArgumentValue.kt:121)
        at com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.convertValue(convertArgumentValue.kt:76)
        at com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.convertArgumentValue(convertArgumentValue.kt:54)
        at com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.mapToKotlinObject(convertArgumentValue.kt:119)
        at com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.convertValue(convertArgumentValue.kt:76)
        at com.expediagroup.graphql.generator.execution.ConvertArgumentValueKt.convertArgumentValue(convertArgumentValue.kt:54)
        at com.expediagroup.graphql.generator.execution.FunctionDataFetcher.mapParameterToValue(FunctionDataFetcher.kt:98)
        at com.expediagroup.graphql.generator.execution.FunctionDataFetcher.getParameters(FunctionDataFetcher.kt:78)
        at com.expediagroup.graphql.generator.execution.FunctionDataFetcher.get(FunctionDataFetcher.kt:57)
        at suwayomi.tachidesk.graphql.directives.RequireAuthDirectiveWiring.onField$lambda$0(RequireAuthDirectiveWiring.kt:44)
        at graphql.execution.ExecutionStrategy.invokeDataFetcher(ExecutionStrategy.java:533)
        at graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:497)
        at graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:438)
        at graphql.execution.ExecutionStrategy.resolveFieldWithInfo(ExecutionStrategy.java:397)
        at graphql.execution.ExecutionStrategy.getAsyncFieldValueInfo(ExecutionStrategy.java:335)
        at graphql.execution.AsyncExecutionStrategy.execute(AsyncExecutionStrategy.java:57)
        at graphql.execution.Execution.executeOperation(Execution.java:181)
        at graphql.execution.Execution.execute(Execution.java:117)
        at graphql.GraphQL.execute(GraphQL.java:546)
        at graphql.GraphQL.lambda$parseValidateAndExecute$13(GraphQL.java:476)
        at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
        at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2341)
        at graphql.GraphQL.parseValidateAndExecute(GraphQL.java:471)
        at graphql.GraphQL.lambda$executeAsync$9(GraphQL.java:429)
        at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
        at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2341)
        at graphql.GraphQL.executeAsync(GraphQL.java:418)
        at com.expediagroup.graphql.server.execution.GraphQLRequestHandler.execute(GraphQLRequestHandler.kt:103)
        at com.expediagroup.graphql.server.execution.GraphQLRequestHandler.executeRequest$suspendImpl(GraphQLRequestHandler.kt:82)
        at com.expediagroup.graphql.server.execution.GraphQLRequestHandler.executeRequest(GraphQLRequestHandler.kt)
        at com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invokeSuspend(GraphQLServer.kt:62)
        at com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invoke(GraphQLServer.kt)
        at com.expediagroup.graphql.server.execution.GraphQLServer$execute$2.invoke(GraphQLServer.kt)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:43)
        at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:286)
        at com.expediagroup.graphql.server.execution.GraphQLServer.execute$suspendImpl(GraphQLServer.kt:49)
        at com.expediagroup.graphql.server.execution.GraphQLServer.execute(GraphQLServer.kt)
        at suwayomi.tachidesk.graphql.controller.GraphQLController$execute$1$1.invokeSuspend(GraphQLController.kt:26)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
        at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
Caused by: java.lang.ClassCastException: Cannot cast java.lang.Double to java.lang.Float
        at java.base/java.lang.Class.cast(Class.java:4067)
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        ... 52 common frames omitted
```

### Suwayomi-Server version

v2.1.1972

### Used database

H2

### Server operating system

Ubuntu 24.04

### Server Desktop Environment

-

### Server JVM version

Zulu21.44+17-CA

### Used client name

-

### Client version

-

### Used web browser

-

### Client operating system

-

### Other details

_No response_

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [x] I have written a short but informative title (ideally less than ~100 characters).
- [x] I have tried the troubleshooting guide described in [README.md](https://github.com/Suwayomi/Suwayomi-Server?tab=readme-ov-file#troubleshooting-and-support)
- [x] I have updated to the **[latest version](https://github.com/suwayomi/suwayomi-server/releases/latest)**.
- [x] I have filled out all of the requested information in this form, including specific version numbers.
- [x] I understand that **Suwayomi does not have or fix any extensions**, and I **will not receive help** for any issues related to sources or extensions.

--- a/server/src/main/kotlin/suwayomi/tachidesk/graphql/server/TachideskGraphQLSchema.kt
+++ b/server/src/main/kotlin/suwayomi/tachidesk/graphql/server/TachideskGraphQLSchema.kt
@@ -46,6 +46,7 @@ import suwayomi.tachidesk.graphql.queries.UpdateQuery
 import suwayomi.tachidesk.graphql.server.primitives.Cursor
 import suwayomi.tachidesk.graphql.server.primitives.GraphQLCursor
 import suwayomi.tachidesk.graphql.server.primitives.GraphQLDurationAsString
+import suwayomi.tachidesk.graphql.server.primitives.GraphQLFloatAsDouble
 import suwayomi.tachidesk.graphql.server.primitives.GraphQLLongAsString
 import suwayomi.tachidesk.graphql.server.primitives.GraphQLUpload
 import suwayomi.tachidesk.graphql.subscriptions.DownloadSubscription
@@ -64,6 +65,7 @@ class CustomSchemaGeneratorHooks : FlowSubscriptionSchemaGeneratorHooks() {
     override fun willGenerateGraphQLType(type: KType): GraphQLType? =
         when (type.classifier as? KClass<*>) {
             Long::class -> GraphQLLongAsString // encode to string for JS
+            Float::class -> GraphQLFloatAsDouble // encode float as double
             Duration::class -> GraphQLDurationAsString // encode Duration as ISO-8601 string
             Cursor::class -> GraphQLCursor
             UploadedFile::class -> GraphQLUpload

--- a/server/src/main/kotlin/suwayomi/tachidesk/graphql/server/primitives/FloatAsDouble.kt
+++ b/server/src/main/kotlin/suwayomi/tachidesk/graphql/server/primitives/FloatAsDouble.kt
@@ -33,21 +33,18 @@ private class GraphqlActualFloatCoercing : Coercing<Float, Double> {
     private fun parseValueImpl(
         input: Any,
         locale: Locale,
-    ): Float {
-        if (input is Int) {
-            return input.toFloat()
-        }
-        if (input !is Double) {
-            throw CoercingParseValueException(
+    ): Float =
+        when (input) {
+            is Int -> input.toFloat()
+            is Double -> input.toFloat()
+            else -> throw CoercingParseValueException(
                 CoercingUtil.i18nMsg(
                     locale,
-                    "String.unexpectedRawValueType",
+                    "Float.unexpectedRawValueType",
                     CoercingUtil.typeName(input),
                 ),
             )
         }
-        return input.toFloat()
-    }
 
     private fun parseLiteralImpl(
         input: Any,
@@ -58,7 +55,7 @@ private class GraphqlActualFloatCoercing : Coercing<Float, Double> {
                 CoercingUtil.i18nMsg(
                     locale,
                     "Scalar.unexpectedAstType",
-                    "StringValue",
+                    "FloatValue",
                     CoercingUtil.typeName(input),
                 ),
             )
@@ -73,7 +70,7 @@ private class GraphqlActualFloatCoercing : Coercing<Float, Double> {
                 toDoubleImpl(input) ?: throw CoercingSerializeException(
                     CoercingUtil.i18nMsg(
                         Locale.getDefault(),
-                        "String.unexpectedRawValueType",
+                        "Float.unexpectedRawValueType",
                         CoercingUtil.typeName(input),
                     ),
                 ),
@@ -84,7 +81,7 @@ private class GraphqlActualFloatCoercing : Coercing<Float, Double> {
         toDoubleImpl(dataFetcherResult) ?: throw CoercingSerializeException(
             CoercingUtil.i18nMsg(
                 Locale.getDefault(),
-                "String.unexpectedRawValueType",
+                "Float.unexpectedRawValueType",
                 CoercingUtil.typeName(dataFetcherResult),
             ),
         )
@@ -98,7 +95,7 @@ private class GraphqlActualFloatCoercing : Coercing<Float, Double> {
         toDoubleImpl(dataFetcherResult) ?: throw CoercingSerializeException(
             CoercingUtil.i18nMsg(
                 locale,
-                "String.unexpectedRawValueType",
+                "Float.unexpectedRawValueType",
                 CoercingUtil.typeName(dataFetcherResult),
             ),
         )

--- a/server/src/main/kotlin/suwayomi/tachidesk/graphql/server/primitives/FloatAsDouble.kt
+++ b/server/src/main/kotlin/suwayomi/tachidesk/graphql/server/primitives/FloatAsDouble.kt
@@ -1,0 +1,135 @@
+package suwayomi.tachidesk.graphql.server.primitives
+
+import graphql.GraphQLContext
+import graphql.execution.CoercedVariables
+import graphql.language.FloatValue
+import graphql.language.Value
+import graphql.scalar.CoercingUtil
+import graphql.schema.Coercing
+import graphql.schema.CoercingParseLiteralException
+import graphql.schema.CoercingParseValueException
+import graphql.schema.CoercingSerializeException
+import graphql.schema.GraphQLScalarType
+import java.util.Locale
+
+val GraphQLFloatAsDouble: GraphQLScalarType =
+    GraphQLScalarType
+        .newScalar()
+        .name(
+            "Float32",
+        ).description("32-bit float aka kotlin.lang.Float")
+        .coercing(GraphqlActualFloatCoercing())
+        .build()
+
+private class GraphqlActualFloatCoercing : Coercing<Float, Double> {
+    private fun toDoubleImpl(input: Any): Double? =
+        when (input) {
+            is Float -> input.toDouble()
+            is Double -> input
+            is Int -> input.toDouble()
+            else -> null
+        }
+
+    private fun parseValueImpl(
+        input: Any,
+        locale: Locale,
+    ): Float {
+        if (input is Int) {
+            return input.toFloat()
+        }
+        if (input !is Double) {
+            throw CoercingParseValueException(
+                CoercingUtil.i18nMsg(
+                    locale,
+                    "String.unexpectedRawValueType",
+                    CoercingUtil.typeName(input),
+                ),
+            )
+        }
+        return input.toFloat()
+    }
+
+    private fun parseLiteralImpl(
+        input: Any,
+        locale: Locale,
+    ): Float {
+        if (input !is FloatValue) {
+            throw CoercingParseLiteralException(
+                CoercingUtil.i18nMsg(
+                    locale,
+                    "Scalar.unexpectedAstType",
+                    "StringValue",
+                    CoercingUtil.typeName(input),
+                ),
+            )
+        }
+        return input.value.toFloat()
+    }
+
+    private fun valueToLiteralImpl(input: Any): FloatValue =
+        FloatValue
+            .newFloatValue()
+            .value(
+                toDoubleImpl(input) ?: throw CoercingSerializeException(
+                    CoercingUtil.i18nMsg(
+                        Locale.getDefault(),
+                        "String.unexpectedRawValueType",
+                        CoercingUtil.typeName(input),
+                    ),
+                ),
+            ).build()
+
+    @Deprecated("")
+    override fun serialize(dataFetcherResult: Any): Double =
+        toDoubleImpl(dataFetcherResult) ?: throw CoercingSerializeException(
+            CoercingUtil.i18nMsg(
+                Locale.getDefault(),
+                "String.unexpectedRawValueType",
+                CoercingUtil.typeName(dataFetcherResult),
+            ),
+        )
+
+    @Throws(CoercingSerializeException::class)
+    override fun serialize(
+        dataFetcherResult: Any,
+        graphQLContext: GraphQLContext,
+        locale: Locale,
+    ): Double =
+        toDoubleImpl(dataFetcherResult) ?: throw CoercingSerializeException(
+            CoercingUtil.i18nMsg(
+                locale,
+                "String.unexpectedRawValueType",
+                CoercingUtil.typeName(dataFetcherResult),
+            ),
+        )
+
+    @Deprecated("")
+    override fun parseValue(input: Any): Float = parseValueImpl(input, Locale.getDefault())
+
+    @Throws(CoercingParseValueException::class)
+    override fun parseValue(
+        input: Any,
+        graphQLContext: GraphQLContext,
+        locale: Locale,
+    ): Float = parseValueImpl(input, locale)
+
+    @Deprecated("")
+    override fun parseLiteral(input: Any): Float = parseLiteralImpl(input, Locale.getDefault())
+
+    @Throws(CoercingParseLiteralException::class)
+    override fun parseLiteral(
+        input: Value<*>,
+        variables: CoercedVariables,
+        graphQLContext: GraphQLContext,
+        locale: Locale,
+    ): Float = parseLiteralImpl(input, locale)
+
+    @Deprecated("")
+    override fun valueToLiteral(input: Any): Value<*> = valueToLiteralImpl(input)
+
+    override fun valueToLiteral(
+        input: Any,
+        graphQLContext: GraphQLContext,
+        locale: Locale,
+    ): Value<*> = valueToLiteralImpl(input)
+}

